Flickr Widget: Adds option to open Flickr images in a new tab

chickenn00dle opened the pull request:
Fixes #11405

#### Changes proposed in this Pull Request:
* Previously, Flickr widget images could only be opened in the same window which directs users away from the site owners' site. This PR adds an "Open images in new tab?" option to Flickr widget settings, allowing site owners to choose whether their users are directed away from their site or not.

#### Testing instructions:

1. Activate Jetpack Widgets
2. Add the Flickr (Jetpack) widget to a widget area of your choosing
3. Along with the other settings, check the "Open images in new tab?" checkbox
4. View the Flickr gallery in the front-end and open one of the images.
5. Uncheck the setting to verify images can open in the same window as well

#### Proposed changelog entry for your changes:
* Add option to open Flickr gallery images in new tabs

chickenn00dle committed:
Flickr Widget: Adds option to open Flickr images in a new tab

Addresses this issue: https://github.com/Automattic/jetpack/issues/11405

Previously, the Flickr widget images could only be opened via the same window which
directs userss away from the site owners' site. This PR adds an
"Open images in new tab?" option to flickr widget settings, allowing
site owners to choose whether their users are directed away from their
site or not.

jeherve reviewed (changes requested):
I would recommend setting up your IDE so it lints code as you work; that will avoid issues with the use of spaces instead of tabs in this PR, for example.

![image](https://user-images.githubusercontent.com/426388/61786506-21b81c80-ae0e-11e9-965e-36acd99cc0ae.png)

You can find some instructions here:
https://github.com/Automattic/jetpack/blob/master/docs/development-environment.md#use-php-codesniffer-and-eslint-to-make-sure-your-code-respects-coding-standards

Alternatively, if you run `yarn build` most the necessary tools should become available to you.

chickenn00dle committed:
Replaces opening spaces with tabs

Previous commit was not checked with phpcs and contained several lines
tabbed with spaces. This replaces those spaces with tabs.

chickenn00dle commented:
Ah, apologies!

Looks like I got ahead of myself there 😅

Followed up and made sure there were no issues with PHP CodeSniffer.

jeherve commented on modules/widgets/flickr/form.php:
```diff
@@ -51,6 +51,16 @@ class="widefat"
+	<label>
+		<?php esc_html_e( 'Open images in new tab?', 'jetpack' ); ?>
+	</label>
+	<input
```
I would place the checkbox before the label, I think it feels more natural.

chickenn00dle replied:
Good call. Actually reviewing some of the other checkboxes in other Jetpack widgets it looks like the checkbox and label text is nested within label tags. Made the change for this to be in-line with this pattern.

jeherve reviewed (changes requested):
This tests well. I would only have one minor recommendation for the look of the checkbox.

chickenn00dle committed:
Places checkbox input element before label text

This commit places the checkbox input element for the "Open images in
new tab?" option" before the label text and nested within label tags to
be in line with similar Jetpack widget settings.

jeherve reviewed (approved):
This looks good to me! It should be good to merge after a second review!
